### PR TITLE
Added error handling and feedback feature to lessons form

### PR DESCRIPTION
**Pull request by iyerm2000**
Added the following:

1. Pop-up for when the user does not add any lesson notes. Added some text to ensure that user knows not to leave the field blank
2. Error handling in javascript to give feedback to users when any of their action fails (adding lesson, deleting lesson etc.)

### CODE REVIEW

File: app-main/public/js/quill_init.js [hunk 1]
```diff
@@ -10,6 +10,6 @@ export const quill = new Quill("#editor", {
   modules: {
     toolbar: toolbarOptions,
   },
-  placeholder: "Add some lesson notes!",
+  placeholder: "Add some lesson notes! (Cannot Leave this blank!)",
```
Comment (jrobind) [inline app-main/public/js/quill_init.js]: I like this idea, but I think `Add some lesson notes!` may be clear enough for the user. The alert you've added will trigger if the user tries to submit a lesson without content anyway. 

I think we can probably remove this.

**Review by jrobind — CHANGES_REQUESTED** [review]
This looks great! Thanks for submitting a PR. I think eventually we should try to move away from using alert boxes. There's a few reasons for this:

* They can be very distracting for the user and actually break up user flow (since it blocks the UI thread and also forces user interaction)
* Very difficult to style

Here's some useful [insights and opinions on this](https://stackoverflow.com/questions/20359807/why-shouldnt-i-use-alert-in-javascript?)

This is cool for now, but I think we can raise a ticket to look at another solution in the future (one that is a little more UX friendly).

If you can please remove the placeholder text then I can get this merged in 😄

File: app-main/public/js/quill_init.js (hunk 1 above)
Comment (iyerm2000) [reply app-main/public/js/quill_init.js]: That makes sense! I have made the requested changes. The read on why using alert boxes is not UX friendly makes sense and I will definitely keep that  in mind for my next few projects :)

**Comment by jrobind** [comment]
Thanks for making the change :)

**Pull request merged**